### ISSUE TITLE
Remove unnecessary placeholders on /settings pages

### ISSUE THREAD

**Issue by quinnkeast** [sourcegraph/sourcegraph#21868]
Our content guidelines for actionable language specify that [placeholders should be avoided except in specific situations](https://about.sourcegraph.com/handbook/communication/content_guidelines/actionable_language#placeholders).

There's a few places where we should remove placeholders altogether, or switch to using a helper.

- [ ] On the /settings/profile page, the display name and avatar URL fields have placeholders that duplicate the label. These should be removed and the field left empty.

<img width="451" alt="CleanShot 2021-06-08 at 14 37 18@2x" src="https://user-images.githubusercontent.com/6304497/121186124-0a01aa00-c867-11eb-9a9c-2319fbcf5b25.png">

- [ ] On the /settings/tokens/new page, the "What's this token for?" placeholder should be removed and instead placed as helper text.

<img width="507" alt="CleanShot 2021-06-08 at 14 41 27@2x" src="https://user-images.githubusercontent.com/6304497/121186658-9dd37600-c867-11eb-8a62-2a1ceb6cc0de.png">
 
CC @AlicjaSuska

**Comment by AlicjaSuska** [comment]
Thank you for bringing this up @quinnkeast. In general, I think that leaving fields empty looks like a bug to many users as people are so used to seeing them, especially in short forms like this.

I think we should revisit this guideline. I agree that placeholder shouldn't be repeating the label though.

In this specific case I'd use the following placeholders (to explain the labels even better):
- Display name: Your name visible to other Sourcegraph users
- Avatar URL: Link to your profile picture

@quinnkeast wdyt about this proposition?

**Comment by quinnkeast** [comment]
I disagree on this one from a usability heuristic perspective. There's a [fair bit of stuff out there](https://www.smashingmagazine.com/2018/06/placeholder-attribute/) on [placeholders being harmful for usability](https://www.nngroup.com/articles/form-design-placeholders/) or otherwise not great for accessibility. The redesign magnifies this, as the colour of the placeholders is easily mistaken as content in an already-filled field.

On a personal anecdotal level, I've found far more trouble in previous user testing with users thinking fields are already populated, than that there's a bug.

That people are used to seeing a bad pattern, in this case, feels like something we should challenge rather than follow.

**Comment by AlicjaSuska** [comment]
Maybe the problem is the placeholder color in this case? 🤔
